lib/account.rb

```diff
@@ -0,0 +1,66 @@
+require 'csv'
+require 'date'
+require_relative 'owner'
+
+module Bank
+  class Account
+    attr_reader :id, :balance, :date, :owner
+
+    def initialize(id, balance, date)
+      raise ArgumentError.new "ID must be an Integer" if id.class != Integer
+      @id = id
+
+      if balance < 0 || balance.class != Integer
+        raise ArgumentError.new "Balance must be a non-negative number"
+      end
+      @balance = balance
+
+      @date = DateTime.parse(date)
+
+      CSV.read("support/account_owners.csv").each do |line|
```

Comment:
This loop looks like a good candidate for a method.  Maybe an `Owner.find` method!